analyses/fusion-frequencies/01-fusion-frequencies.R

```diff
@@ -0,0 +1,287 @@
+
+
+# Subset independent samples in histology table --------------------------------
+message('Prepare data...')
+# td = tumor descriptor
+td_htl_dfs <- list(
+  primary_htl_df = htl_df %>%
+    filter(Kids_First_Biospecimen_ID %in%
+             primary_indp_sdf$Kids_First_Biospecimen_ID),
+  
+  relapse_htl_df = htl_df %>%
+    filter(Kids_First_Biospecimen_ID %in%
+             relapse_indp_sdf$Kids_First_Biospecimen_ID),
+  
+  overall_htl_df = htl_df %>%
+    filter(sample_type == 'Tumor')
+)
+
+# keep only samples with non-NA cancer_group and cohort
+# add cancer_group_cohort column
+td_htl_dfs <- lapply(td_htl_dfs, function(x) {
+  # assert all Kids_First_Biospecimen_IDs are unique
+  stopifnot(identical(nrow(x), length(unique(x$Kids_First_Biospecimen_ID))))
+  
+  fx <- filter(x, !is.na(cancer_group), !is.na(cohort))
+  return(fx)
+})
+
+# Subset tumor samples to filter alteration df ---------------------------
+tumor_kfbids <- htl_df %>%
+  filter(sample_type == 'Tumor') %>%
+  pull(Kids_First_Biospecimen_ID)
+
+
+fusion_df <- fusion_df  %>%
+  # We want to keep track of the gene symbols for each sample-fusion pair
+  dplyr::select(Sample,
+                Kids_First_Participant_ID, 
+                FusionName, 
+                Gene1A, 
+                Gene1B, 
+                Gene2A, 
+                Gene2B, 
+                Fusion_Type,
+                DomainRetainedGene1A,
+                DomainRetainedGene1B,
+                reciprocal_exists,
+                annots,
+                BreakpointLocation,
+                ends_with("anno")) %>%
+  # Update reciprocal_exits to fusion with atleast 1 kinase gene involved
+  mutate(
+         reciprocal_exists_kinase = 
+           # if either gene are kinase we will have "Yes" or "No" values
+           # in DomainRetainedGene1A and DomainRetainedGene1B
+           if_else(is.na(DomainRetainedGene1A) &
+                     is.na(DomainRetainedGene1B) ,
+                   # reciprocl_exists is a logical column
+                   FALSE,
+                   reciprocal_exists)) %>%
+  dplyr::rename("Kids_First_Biospecimen_ID"="Sample",
+         "Kinase_domain_retained_Gene1A" = "DomainRetainedGene1A",
+         "Kinase_domain_retained_Gene1B" = "DomainRetainedGene1B",
+         "Reciprocal_exists_either_gene_kinase" = "reciprocal_exists_kinase")%>%
+  # replace NA to "" in columns that have NA
+  replace_na(list("Kinase_domain_retained_Gene1A"="",
+                  "Kinase_domain_retained_Gene1B"="",
+                  "Reciprocal_exists_either_gene_kinase"="",
+                  "Gene1A_anno"="",
+                  "Gene1B_anno"="",
+                  "Gene2A_anno"="",
+                  "Gene2B_anno"="",
+                  "Fusion_anno"=""))%>%
+  # We want a single column that contains the gene symbols
+  tidyr::gather(Gene1A, Gene1B, Gene2A, Gene2B,
+                key = gene_position, value = gene_symbol) %>%
+  filter(!is.na(gene_symbol)) %>%
```

Comment:
@jharenza @kgaonkar6 I looked into the issue that some gene symbols have no mapped ENSG ID as @kgaonkar6 pointed out at <https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/49#issuecomment-889450851>. The following gene symbols are the only ones that have no ENSG ID: IGH-@, IGH@, IGL-@, and IGL@. (The code for checking them is listed in the end of this comment.) These gene symbols are [IGL immunoglobulin lambda locus](https://www.ncbi.nlm.nih.gov/gene?Db=gene&Cmd=DetailsSearch&Term=3535) and [IGH immunoglobulin heavy locus](https://www.ncbi.nlm.nih.gov/gene/3492), and there is no Ensembl link out from the NCBI gene pages. These gene symbols are introduced by starfusion, <https://github.com/STAR-Fusion/STAR-Fusion/issues/197#issuecomment-597215303>, but I am not sure if there is any difference between IGH-@ and IGH@, or between IGL-@ and IGL@.

These two loci are relevant to lymphoma, according to <https://www.sciencedirect.com/topics/biochemistry-genetics-and-molecular-biology/igl> and <https://www.sciencedirect.com/topics/biochemistry-genetics-and-molecular-biology/igh>, so I think we need to keep them searchable in the PedOT database.

I have been thinking about the following possible solutions:

- If starfusion also outputs actual IGH/IGL genes fusions, e.g. [IGHE ENSG00000211891](https://useast.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000211891;r=14:105597691-105601728) as additional rows, maybe we could remove these symbols. Also add note and code to check these are the only gene symbols without ENSG ID, for the ease of maintenance in the future. If starfusion does not output corresponding IGH/IGL genes, could we infer the genes from the break point or the other fused gene?
- If these IGH/L@ fusions really have no corresponding ENSG IDs and are useful, maybe we could discuss with the FNL team on handling these gene symbols that have no ENSG ID on the PedOT website in the `#ot-portal-content` channel.

```text
> f_tbl <- read_tsv('../../fusion-frequencies/results/putative-oncogene-fusion-freq.tsv.gz')
> f_gene_tbl <- read_tsv('../../fusion-frequencies/results/putative-oncogene-fused-gene-freq.tsv.gz')
> map_tbl <- read_tsv('../../../data/ensg-hugo-rmtl-mapping.tsv')
> f_tbl %>% filter(!Gene_Symbol %in% map_tbl$gene_symbol) %>% group_by(Gene_Symbol) %>% tally()
# A tibble: 4 x 2
  Gene_Symbol     n
  <chr>       <int>
1 IGH-@           3
2 IGH@           10
3 IGL-@           2
4 IGL@            2
> f_gene_tbl %>% filter(!Gene_Symbol %in% map_tbl$gene_symbol) %>% group_by(Gene_Symbol) %>% tally()
# A tibble: 4 x 2
  Gene_Symbol     n
  <chr>       <int>
1 IGH-@           9
2 IGH@           44
3 IGL-@           4
4 IGL@            4
```

Reply:
Thanks @logstar from bringing up this issue.

We've noticed this issue with IGH@ symbols before and had to manually retain/annotate these fusions with IGH*@ as oncogenic fusions in OpenPBTA and current `fusion_filtering` modules. 

When we checked (last year I think) there were multiple IGH genes around the breakpoint and gtf, we don't have the IGH*@ genes in the gtf file we use to get the ENSG ID so you are right will have to discuss and manually add the matched ENSG ID to `ensg-hugo-rmtl-mapping.tsv`.

Reply:
> Thanks @logstar from bringing up this issue.
> 
> We've noticed this issue with IGH@ symbols before and had to manually retain/annotate these fusions with IGH*@ as oncogenic fusions in OpenPBTA and current `fusion_filtering` modules.
> 
> When we checked (last year I think) there were multiple IGH genes around the breakpoint and gtf, we don't have the IGH*@ genes in the gtf file we use to get the ENSG ID so you are right will have to discuss and manually add the matched ENSG ID to `ensg-hugo-rmtl-mapping.tsv`.

@kgaonkar6 Thank you for brining more context to this issue.

Regarding your suggested solution to manually add the matched ENSG IDs, I was wondering if there is any ENSG ID that directly maps to the IG*@ symbols. I checked NCBI gene, HGNC, mygene.info, and biomart, and there is no direct mapping from IGH/IGL to ENSG ID.

@jharenza and @kgaonkar6 Therefore, I was wondering if we should defer this issue as a fusion filtering enhancement and do not handle them in this `fusion-frequencies` module.

- IGH
  - NCBI gene: https://www.ncbi.nlm.nih.gov/gene/3492
  - mygene.info: http://mygene.info/v3/gene/3492
  - HGNC: https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:5477
  - biomart: see code below
- IGL
  - NCBI gene: https://www.ncbi.nlm.nih.gov/gene?Db=gene&Cmd=DetailsSearch&Term=3535
  - mygene.info: http://mygene.info/v3/gene/3535
  - HGNC: https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:5853
  - biomart: see code below

```text
> bm_ensembl <- biomaRt::useEnsembl(
+   biomart = 'ensembl', dataset = 'hsapiens_gene_ensembl')
> biomaRt::getBM(
+   attributes = c('ensembl_gene_id', 'description'),
+   filters = c('hgnc_id'),
+   values = c(
+     'HGNC:5853', # IGL
+     'HGNC:5477', # IGH
+     'HGNC:5876' # this should have ENSG ID, to show this code works
+     ),
+   mart = bm_ensembl)
  ensembl_gene_id                                                            description
1 ENSG00000273737 immunoglobulin lambda variable 1-36 [Source:HGNC Symbol;Acc:HGNC:5876]
2 ENSG00000211655 immunoglobulin lambda variable 1-36 [Source:HGNC Symbol;Acc:HGNC:5876]
```

Reply:
>@jharenza and @kgaonkar6 Therefore, I was wondering if we should defer this issue as a fusion filtering enhancement and do not handle them in this fusion-frequencies module.

Yes, we can do that to get this PR merged.

Reply:
> > @jharenza and @kgaonkar6 Therefore, I was wondering if we should defer this issue as a fusion filtering enhancement and do not handle them in this fusion-frequencies module.
> 
> Yes, we can do that to get this PR merged.

Thank you. I will submit a fusion-filtering ticket summarizing our discussions here and label the ticket with enhancement and low-priority.

I will resolve this conversation shortly.